src/NuGet.Services.AzureSearch/SearchDocumentBuilder.cs

```diff
@@ -361,18 +363,94 @@ private static void PopulateKey(KeyedDocument document, string packageId, Search
             document.IsExcludedByDefault = isExcludedByDefault;
         }
 
-        private static void PopulateDeprecation(
+        private static void PopulateDeprecationFromDb(
             SearchDocument.Full document,
             Package package)
+        {
+            if (package.Deprecations?.Count != 1)
+            {
+                return;
+            }
+
+            var packageDeprecation = package.Deprecations?.ElementAt(0) as Entities.PackageDeprecation;
+            if (packageDeprecation == null || packageDeprecation.Status == PackageDeprecationStatus.NotDeprecated)
             {
-                document.Deprecation = new Deprecation(); //TODO: https://github.com/NuGet/NuGetGallery/issues/7297
+                return;
             }
 
-        private static void PopulateVulnerabilities(
+            var version = packageDeprecation.AlternatePackage?.Version ?? "";
+            document.Deprecation = new Deprecation()
+            {
+                Message = packageDeprecation.CustomMessage,
+                Reasons = packageDeprecation.Status.ToString().Replace(" ", "").Split(','),
+                AlternatePackage = packageDeprecation.AlternatePackage == null ? null : new AlternatePackage()
+                {
+                    Id = packageDeprecation.AlternatePackage.Id,
+                    Range = $"[{version}, )"
+                }
+            };
+        }
+
+        private static void PopulateDeprecationFromCatalog(
+            SearchDocument.UpdateLatest document,
+            PackageDetailsCatalogLeaf leaf)
+        {
+            if (leaf.Deprecation?.Reasons == null || !leaf.Deprecation.Reasons.Any())
+            {
+                return;
+            }
+
+            document.Deprecation = new Deprecation()
+            {
+                Reasons = leaf.Deprecation.Reasons.ToArray<string>(),
```

Comment:
```suggestion
                Reasons = leaf.Deprecation.Reasons.ToArray(),
```
The type parameter should be inferred automatically.